os/drivers/audio/alc5658.c

```diff
@@ -1464,21 +1505,23 @@ FAR struct audio_lowerhalf_s *alc5658_initialize(FAR struct i2c_dev_s *i2c, FAR
 	/* Allocate a ALC5658 device structure */
 	priv = (FAR struct alc5658_dev_s *)kmm_zalloc(sizeof(struct alc5658_dev_s));
 
-	if (!priv) {
-		return NULL;
-	}
-
 	/* Initialize the ALC5658 device structure.  Since we used kmm_zalloc,
 	 * only the non-zero elements of the structure need to be initialized.
 	 */
 	priv->dev.ops = &g_audioops;
 	priv->lower = lower;
 	priv->i2c = i2c;
 	priv->i2s = i2s;
+	priv->gain = ALC5658_GAIN_DEFAULT;
+	priv->volume = ALC5658_HP_VOL_DEFAULT;
 
 	sem_init(&priv->devsem, 0, 1);
 	sq_init(&priv->pendq);
 
+	if (!priv) {
```

Comment:
Please check this you already used priv before check this.

Reply:
Moved.